### PR TITLE
Refactor utils.getWebSocket to improve test coverage by +0.2%

### PR DESCRIPTION
**Pull request by lucasrmendonca**
### Description, Motivation and Context
- Refactor `utils.getWebSocket` so it's easier to unit test;
- Move logic from `utils.getWebSocket` that checked for the browser's Window Object into a new function `utils.getWebSocketFromWindowObject`;
- Add jsdoc comments to functions `utils.getWebSocket` and `utils.getWebSocketFromWindowObject`;

### What kind of change does this PR introduce?
- [x] Refactoring (no new functionality, only code improvements)

### Checklist:
- [x] My code follows the code style of this project.
- [x] I have added tests to cover my changes.
- [x] Overall test coverage is not decreased.
- [x] All new and existing tests passed.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by lucasrmendonca** [comment]
Although code climate is complaining, this change fixed 4 code climate issues while introducing 1 new one, so it's still an improvement

File: src/utils.js [hunk 1]
```diff
@@ -21,29 +21,34 @@ export function getServerUrlForBrowser (url) {
...
     }
 
     // assuming just domain + path
     return `${scheme}${url}`;
 }
 
-export function getWebSocket (url, protocols, ws, headers, requestOptions) {
-    const parsedUrl = isNode ? getServerUrlForNode(url) : getServerUrlForBrowser(url);
+
+export function getWebSocket (wampy = {}) {
```
Comment (KSDaemon) [inline src/utils.js:33]: As it is not clear from the function definition what options it uses, I think we need at least to describe it in @JSDoc format: prop + format + optional? + desc
example: 
```js
/**
  * @param {Object} userInfo Information about the user.
  * @param {string} userInfo.name The name of the user.
  * @param {string} userInfo.email The email of the user.
  */
 function logIn(userInfo) {
        doLogIn(userInfo.name, userInfo.email);
 }
```

**Review by KSDaemon — CHANGES_REQUESTED** [review]

**Comment by lucasrmendonca** [comment]
I've added the jsdoc comment block and also fixed codeclimate's complaint by creating a helper function just for the case when the WebSocket class is part of the global `window` object, let me know if this is ok

**Comment by KSDaemon** [comment]
Okay. Let's land this. 👍🏻

**Pull request merged**